SDL/Kernels.h

```diff
@@ -490,7 +490,7 @@ namespace SDL
                     if (alpaka::math::abs(acc, eta_pix2 - eta_pix1) > 0.1f)
                         continue;
 
-                    if (secondpass && (!segmentsInGPU.isQuad[jx] || segmentsInGPU.isDup[jx]))
+                    if (secondpass && !segmentsInGPU.isQuad[jx])
```

Comment:
what's the impact of this change on efficiency?

Reply:
Do you mean as in timing? If so, it seems to not be too much as shown by the timings I posted in the first comment.

If you mean reconstruction efficiency, then this is what can break it and make it remove all pLSs.

Reply:
> If you mean reconstruction efficiency, then this is what can break it and make it remove all pLSs.

yes, my question was `what's the impact of this change on efficiency?`

If I understand correctly, the test against `isDup` here includes the first pass marking and removal of the check allows to consider all original pLSs. I'm not sure this is appropriate in the second pass.

if more rigorous dual info (from the first and second pass) is needed, then perhaps a change should be made from `bool* isDup` to `char* isDup` and set the value to 1 in the first pass and 2 in the second

Reply:
Oh yeah, that's a good point. I'll check tomorrow and let's talk about it in the meeting.

Reply:
If you make this change you end up with the same results as here: https://github.com/SegmentLinking/TrackLooper/pull/323#issuecomment-1720059884

I'm completely onboard with leaving the `isDup` check as it is or as a more robust version. What I think we should focus on is that the `npMatched >= 1 or dR2 < 0.00075f and (ix < jx)` statement was wrong and fixing it is very negatively affecting the results.